File: src/main/kotlin/coraythan/keyswap/users/search/UserSearchResult.kt
package coraythan.keyswap.users.search

import coraythan.keyswap.generatets.GenerateTs
import coraythan.keyswap.patreon.PatreonRewardsTier
import coraythan.keyswap.users.UserType
import java.util.*

@GenerateTs
data class UserSearchResult(
        val id: UUID,
        val username: String,
        val rating: Double,
        val deckCount: Int,
        val forSaleCount: Int,
        val topSasAverage: Int,
        val highSas: Int,
        val lowSas: Int,
        val totalPower: Int,
        val totalChains: Int,
        val mavericks: Int,
        val anomalies: Int,
        val role: UserType,
        val patreonTier: PatreonRewardsTier?,
        val manualPatreonTier: PatreonRewardsTier?,
        val teamId: UUID?,
        val allowUsersToSeeDeckOwnership: Boolean = true,
)

File: ui/src/user/search/UserSearchResult.ts
import { PatreonRewardsTier } from "../../thirdpartysites/patreon/PatreonRewardsTier"
import { UserType } from "../KeyUser"

export interface UserSearchResult {
    id: string
    username: string
    rating: number
    deckCount: number
    forSaleCount: number
    patreonTier?: PatreonRewardsTier
    manualPatreonTier?: PatreonRewardsTier
    topSasAverage: number
    highSas: number
    lowSas: number
    totalPower: number
    totalChains: number
    mavericks: number
    anomalies: number
    role: UserType
    teamName?: string
}
